FirebaseRemoteConfig/SwiftNew/ConfigRealtime.swift

```diff
@@ -0,0 +1,662 @@
+                        "Fetched config's template version is outdated, re-fetching")
+            self.autoFetch(attempts: attempts, targetVersion: targetVersion)
+          }
+        } else {
+          RCLog.debug("I-RCN000016",
+                      "Fetched config's template version is outdated, re-fetching")
+          self.autoFetch(attempts: attempts, targetVersion: targetVersion)
+        }
+      }
+    }
+  }
+
+  @objc(scheduleFetch:targetVersion:) public
+  func scheduleFetch(remainingAttempts: Int, targetVersion: Int) {
+    let delay = TimeInterval.random(in: 0 ... 3) // Random delay between 0 and 3 seconds
+    realtimeLockQueue.asyncAfter(deadline: .now() + delay) {
+      self.fetchLatestConfig(remainingAttempts: remainingAttempts, targetVersion: targetVersion)
+    }
+  }
+
+  @objc(autoFetch:targetVersion:) public
+  func autoFetch(attempts: Int, targetVersion: Int) {
+    realtimeLockQueue.async {
+      guard attempts > 0 else {
+        let error = NSError(domain: RemoteConfigUpdateErrorDomain,
+                            code: RemoteConfigUpdateError.notFetched.rawValue,
+                            userInfo: [
+                              NSLocalizedDescriptionKey: "Unable to fetch the latest version of the template.",
+                            ])
+        RCLog.error("I-RCN000011", "Ran out of fetch attempts, cannot find target config version.")
+        self.propagateErrors(error)
+        return
+      }
+      self.scheduleFetch(remainingAttempts: attempts, targetVersion: targetVersion)
+    }
+  }
+
+  // MARK: - URLSessionDataDelegate
+
+  public func urlSession(_ session: URLSession, dataTask: URLSessionDataTask,
+                         didReceive data: Data) {
+    if !session.isEqual(self.session) {
+      return
+    }
+
+    let strData = String(data: data, encoding: .utf8) ?? ""
+    if strData.contains(serverForbiddenStatusCode) {
+      let error = NSError(domain: RemoteConfigUpdateErrorDomain,
+                          code: RemoteConfigUpdateError.streamError.rawValue,
+                          userInfo: [NSLocalizedDescriptionKey: strData])
+      RCLog.error("I-RCN000021", "Cannot establish connection. \(error)")
+      propagateErrors(error)
+      return
+    }
+
+    if let beginRange = strData.range(of: "{"),
+       let endRange = strData.range(of: "}") {
+      RCLog.debug("I-RCN000015", "Received config update message on stream.")
+      let msgRange = Range(uncheckedBounds: (lower: beginRange.lowerBound,
+                                             upper: strData.index(after: endRange.upperBound)))
+      let jsonData = String(strData[msgRange]).data(using: .utf8)!
+      do {
+        if let response = try JSONSerialization.jsonObject(with: jsonData,
+                                                           options: []) as? [String: Any] {
+          evaluateStreamResponse(response)
+        }
+      } catch {
+        propagateErrors(error)
```

Comment:
In ObjC, this error was not propagated directly, but a more generic error was thrown in evaluateStreamResponse:

```objc
    NSError *error =
        [NSError errorWithDomain:FIRRemoteConfigUpdateErrorDomain
                            code:FIRRemoteConfigUpdateErrorMessageInvalid
                        userInfo:@{NSLocalizedDescriptionKey : @"Unable to parse ConfigUpdate."}];
    [self propagateErrors:error];
```
I think the main effect is the error won't be wrapped in the RC domain and code. I think we should instead define an error here that matches the above snippet with an underlying error key in the user info that points to the caught error here.